src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -200,6 +206,14 @@ module.exports = function<T, P, I, TI, C, CX>(config : HostConfig<T, P, I, TI, C
     if (highestPriorityRoot) {
       nextPriorityLevel = highestPriorityLevel;
       priorityContext = nextPriorityLevel;
+
+      // Before we start any new work, let's make sure that we have a fresh
+      // stack to work from.
+      // TODO: This call is burried a bit too deep. It would be nice to have
+      // a single point which happens right before any new work and
+      // unfortunately this is it.
+      resetContextStack();
```

Comment:
We could but I suspect that with #8611 these will be unified to a single call to an external module which is why I grouped them inside of just inlining them.